backend/dataall/core/environment/cdk/pivot_role_stack.py

```diff
@@ -107,22 +110,32 @@ def create_pivot_role(self, principal_id: str, external_id: str) -> iam.Role:
             managed_policies=managed_policies,
         )
 
-        role.assume_role_policy.add_statements(
-            iam.PolicyStatement(
-                effect=iam.Effect.ALLOW,
-                principals=[iam.AccountPrincipal(account_id=principal_id)],
-                actions=['sts:AssumeRole'],
-                conditions={
-                    'StringEquals': {'sts:ExternalId': external_id},
-                    'StringLike': {
-                        'aws:PrincipalArn': [
-                            f'arn:aws:iam::{principal_id}:role/*graphql-role',
-                            f'arn:aws:iam::{principal_id}:role/*awsworker-role',
-                            f'arn:aws:iam::{principal_id}:role/*ecs-tasks-role',
-                        ]
+        if ENVNAME == 'local':
```

Comment:
I don't want to derail your PR but our code is full of these checks. Would you mind creating a bunch of util methods (`is_local` `is_prod` `is_docker` etc)?

Reply:
I'll do on a separate PR, good idea!